File: Examples/SharedFiles/Models/Recipe.m
//
//  Recipe.m
//  NordicCook
//
//  Created by Nahuel Marisi on 17/02/2015.
//  Copyright (c) 2015 TechBrewers. All rights reserved.
//

#import "Recipe.h"
#import "Step.h"
#import "Macros.h"
#import "Ingredient.h"
#import "IngredientsSorter.h"

NSString *const kStarterCategory = @"Starter";
NSString *const kMainCategory = @"Main";
NSString *const kDessertCategory = @"Dessert";
NSString *const kAllRecipesCategory = @"AllRecipes";

@implementation Recipe

#pragma mark - Helpers

+ (NSArray *)categoriesArray
{
    return @[kStarterCategory, kMainCategory, kDessertCategory,
             kAllRecipesCategory];
}

+ (NSString *)categoryForIndex:(int)index
{
    NSArray *categories = [self categoriesArray];
    return categories[index];
}

- (NSArray *)ingredients
{
  NSMutableSet *ingredientsArray = [NSMutableSet new];
  for (Step *step in self.steps) {
    [ingredientsArray addObjectsFromArray:step.ingredients];
  }
  
  NSArray *uniqueIngredientsArray = [ingredientsArray allObjects];
  return [IngredientsSorter sortedIngredientsArrayFromArray:uniqueIngredientsArray];
}

- (BOOL)hasStepImages
{
	return self.stepImagesAvailable.boolValue;
}

#pragma mark - Overrides

- (NSString *)description
{
  return description(
										 SelfKeyPath(name),
                     SelfKeyPath(title),
                     SelfKeyPath(subtitle),
                     SelfKeyPath(category),
                     SelfKeyPath(copyright),
                     SelfKeyPath(ingredients),
										 SelfKeyPath(imagePath),
										 SelfKeyPath(hasStepImages),
										 SelfKeyPath(abstract),
										 SelfKeyPath(tip),
										 );
}

@end

File: Examples/SwiftExample/SwiftExample/Models/Recipe.swift
//
//  Recipe.swift
//  SwiftExample
//
//  Created by Nahuel Marisi on 2015-09-03.
//  Copyright © 2015 TechBrewers. All rights reserved.
//

import Foundation

class Recipe: NSObject {
   
    // Properties stored in the JSON file
    var title: String
    var subtitle: String
    var category: String
    
    var difficulty: Int
    var abstract: String?
    
    var time: Int
    var steps: [Step]
    
    var ingredient: Ingredient?
    
    override init() {
        self.title = ""
        self.subtitle = ""
        self.category = ""
        self.difficulty = 1
        self.steps = []
        self.time = 0
    }
    
   init(title: String, subtitle: String, steps: [Step]? = [],
        category: String? = "Unasigned", difficulty: Int? = 1) {
            self.title = title
            self.subtitle = subtitle
            self.category = category!
            self.difficulty = difficulty!
            self.steps = steps!
            self.time = 0
    }
    
}
 